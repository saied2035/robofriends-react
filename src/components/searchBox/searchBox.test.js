import {shallow} from 'enzyme'
import SearchBox from './searchBox'

describe("SearchBox",() => {
    const onSearchChange= jest.fn()
    const props = {onSearchChange}
    const event ={
        target: {
            value: 'some value'
        }
    }
    const robots=[
        {
            id : 1,
            name: 'saied',
            email: 'saied@gmail.com'
        },
        {
            id : 2,
            name: 'saied1',
            email: 'saied1@gmail.com'
        },
        {
            id : 3,
            name: 'saied2',
            email: 'saied2@gmail.com'
        },
        {
            id : 4,
            name: 'saied3',
            email: 'saied3@gmail.com'
        }
    ]
    const wrapperSearchBox = shallow(<SearchBox {...props}/>)

        it("render searchBox",() => {
            expect(wrapperSearchBox).toMatchSnapshot()
        })
        it("check if change event called properly", () => {
            wrapperSearchBox.find('input').simulate('change', (event,robots))
            expect(onSearchChange).toBeCalledWith((event,robots))
        })           
})