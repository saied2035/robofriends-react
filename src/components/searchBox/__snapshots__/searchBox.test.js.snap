// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox render searchBox 1`] = `
<div
  className="pa3"
>
  <input
    className="pa3 ba b--gold bg-light-yellow"
    onChange={[MockFunction]}
    placeholder="search a robot"
    type="text"
  />
</div>
`;
